### PR TITLE
[Ck tile] Support layernorm one pass

### PR DESCRIPTION
**Pull request by rocking5566**
1. Add one pass pipeline, switch one pass and two pass pipeline according to problem size
2. Fix compile error
3. Support padding

### CODE REVIEW

**Comment by poyenc** [comment]
Since `OnePassLayernorm2dFwd()` and `TwoPassLayernorm2dFwd()` share lots of codes. I think maybe we can merge them into one.

**Comment by rocking5566** [comment]
> Since `OnePassLayernorm2dFwd()` and `TwoPassLayernorm2dFwd()` share lots of codes. I think maybe we can merge them into one.

1c644ec523223823a6980172399d549d420df8d0

File: include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp [hunk 1]
```diff
@@ -274,17 +234,176 @@ struct Layernorm2dFwd
...
+                          GammaBlockWindow& gamma_block_window,
+                          BetaBlockWindow& beta_block_window,
+                          YBlockWindow& y_block_window,
+                          MeanBlockWindow& mean_block_window,
+                          InvStdBlockWindow& inv_std_block_window,
+                          ComputeDataType epsilon,
+                          ck_tile::index_t N) const
+    {
+        // TODO: padding - handle max_count if N % kNPerBlock != 0
+        constexpr auto xDstr      = MakeXBlockTileDistribution();
```
Comment (poyenc) [inline include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: to reduce duplicated code, we may get distribution from `x_block_window.get_tile_distribution()`

File: include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp [hunk 2]
```diff
@@ -129,39 +132,27 @@ struct Layernorm2dFwd
...
-
-        constexpr auto xDstr = MakeXBlockTileDistribution();
-
-        auto x_block_window = make_tile_window(
-            x_m_n, make_tuple(number<kMPerBlock>{}, number<kNPerBlock>{}), {iM, 0}, xDstr);
-
         index_t num_n_tile_iteration = __builtin_amdgcn_readfirstlane(N / kNPerBlock);
 
         // TODO: padding - handle max_count if N % kNPerBlock != 0
+        constexpr auto xDstr      = MakeXBlockTileDistribution();
```
Comment (poyenc) [inline include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: to reduce duplicated code, we may get distribution from `x_block_window.get_tile_distribution()`

**Comment by rocking5566** [comment]
I run the test in following branch of flash attention
https://github.com/ROCm/flash-attention/tree/ck_tile/fix-bwd-compiler
4 platforms in [MI200, MI300] * [ROCm6.1, ROCm6.2] has passed

File: include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp (hunk 2 above)
Comment (rocking5566) [reply include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: b410ef0b80c756f0971a7b8af5896f366be640c6

File: include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp (hunk 1 above)
Comment (rocking5566) [reply include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: b410ef0b80c756f0971a7b8af5896f366be640c6

Comment (rocking5566) [reply include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: remove xDstr now

File: include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp (hunk 2 above)
Comment (rocking5566) [reply include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp]: remove xDstr now

**Review by poyenc — APPROVED** [review]

**Pull request merged**